### ISSUE TITLE
Catastrophic performance loss in 1st epoch

### ISSUE THREAD

**Issue by alexisdrakopoulos** [mlfoundations/open_clip#1093]
Hi, I'm using the following:

python -m open_clip_train.main \
    --save-frequency 1 \
    --zeroshot-frequency 1 \
    --dataset-type webdataset \
    --train-data 'data/dataset_{000..042}.tar'  \
    --train-num-samples 1280000 \
    --warmup 90000 \
    --batch-size=9000 \
    --lr=1e-4 \
    --wd=1e-5 \
    --epochs=10 \
    --workers=6 \
    --model ViT-B-16-SigLIP2 \
    --pretrained webli \
    --siglip \
    --opt timm/Lion \
    --grad-checkpointing

with 224x224 images + detailed text pairs with around 2 million samples.

Here is my image retrieval benchmark with the pre-trained model:

--- Benchmark Results ---
Recall@1  : 17.28%
Recall@5  : 34.82%
Recall@10 : 43.18%
Recall@50 : 62.73%
Recall@100: 70.92%
-------------------------

and here are my results after 1 test epoch:

--- Benchmark Results ---
Recall@1  : 5.26%
Recall@5  : 12.78%
Recall@10 : 16.99%
Recall@50 : 28.55%
Recall@100: 34.69%
------------------------

and here after 2 epochs:

--- Benchmark Results ---
Recall@1  : 6.39%
Recall@5  : 14.86%
Recall@10 : 18.79%
Recall@50 : 31.73%
Recall@100: 38.57%
-------------------------

Should I be using a different optimizer? Should I set WD to 0? any advice is appreciated.

**Comment by alexisdrakopoulos** [comment]
So interestingly, doing the exact same thing but with a lower warmup leads to better performance but still very strange performance:

```
Timestamp             Recall@1    Recall@5    Recall@10   Recall@50   
----------------------------------------------------------------------
2025-07-06,19:55:42   17.31%      34.78%      43.18%      62.75%      
2025-07-06,20:40:08   26.15%      49.50%      59.11%      74.82%      
2025-07-06,21:25:25   29.46%      54.07%      63.55%      79.53%      
2025-07-06,22:10:40   28.07%      52.49%      62.66%      79.15%      
2025-07-06,22:55:54   30.42%      55.86%      66.60%      83.17%      
2025-07-06,23:41:08   26.46%      50.69%      61.43%      80.91%      
2025-07-07,00:26:33   28.27%      52.91%      63.94%      83.03%      
2025-07-07,01:11:56   27.82%      52.77%      64.03%      83.08%      
2025-07-07,01:56:48   25.05%      50.39%      61.22%      81.30%      
2025-07-07,02:41:51   28.07%      54.39%      65.09%      83.91%      
2025-07-07,03:27:11   27.11%      51.86%      62.96%      83.45%      
2025-07-07,04:12:34   27.50%      53.23%      64.44%      85.09%      
2025-07-07,04:57:13   28.44%      54.37%      66.09%      85.40%      
2025-07-07,05:42:29   27.06%      52.75%      63.73%      84.53%  
2025-07-07,07:12:59   27.02%      51.26%      62.50%      83.06%  
2025-07-07,07:58:13   29.01%      55.29%      66.51%      85.05%      
2025-07-07,08:42:50   26.23%      51.56%      62.73%      83.50%     
----------------------------------------------------------------------
```

**Comment by rwightman** [comment]
@alexisdrakopoulos don't think this is a bug of any sort. FYI warmup is in steps so if you have a small dataset that could be a lot of epochs, so it means you're LR is way too high and things collapse after the LR hits the breaking point. 1e-4 would be a high fine-tune for AdamW but even higher for LION since I vaguely recall the recommendation for that optimizer being ~1/3 to 1/10 of Adam equivalent?

**Comment by rwightman** [comment]
Moving to discussions...
